src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php

```diff
@@ -125,7 +128,7 @@ public function preSubmit(FormEvent $event)
                 if (!$form->has($name)) {
                     $form->add($name, $this->type, array_replace(array(
                         'property_path' => '['.$name.']',
-                    ), $this->options));
+                    ), $this->getOptionsForEntry($value)));
```

Comment:
I agree with don't misunderstood me, I would assume default view data is string when not null, or when compound an array or object.

In fact, I mentioned that `ChoiceType` is particular in this sense since it is always compound (whether it can be a select input with no children) and will always have a string view data.
That what I would like to ensure decoupling the data transformers from the list mapper in #18180.